pandas/core/window/ewm.py

```diff
@@ -168,20 +185,50 @@ def __init__(
         obj,
         com: Optional[float] = None,
         span: Optional[float] = None,
-        halflife: Optional[float] = None,
+        halflife: Optional[Union[float, TimedeltaConvertibleTypes]] = None,
         alpha: Optional[float] = None,
         min_periods: int = 0,
         adjust: bool = True,
         ignore_na: bool = False,
         axis: int = 0,
+        times: Optional[Union[str, np.ndarray, FrameOrSeries]] = None,
     ):
+        self.com: Optional[float]
         self.obj = obj
-        self.com = get_center_of_mass(com, span, halflife, alpha)
         self.min_periods = max(int(min_periods), 1)
         self.adjust = adjust
         self.ignore_na = ignore_na
         self.axis = axis
         self.on = None
+        if times is not None:
+            if isinstance(times, str):
+                times = self._selected_obj[times]
+            if not is_datetime64_ns_dtype(times):
+                raise ValueError("times must be datetime64[ns] dtype.")
+            if len(times) != len(obj):
+                raise ValueError("times must be the same length as the object.")
+            if not isinstance(halflife, (str, datetime.timedelta)):
+                raise ValueError(
+                    "halflife must be a string or datetime.timedelta object"
+                )
+            self.time_weights = (
+                np.asarray(times - times[0]).astype(np.float64)
+                / Timedelta(halflife).value
+            )
+            # Halflife is no longer applicable when calculating COM
+            # But allow COM to still be calculated if the user passes other decay args
+            if common.count_not_none(com, span, alpha) > 0:
```

Comment:
do we need to do this?

Reply:
We want to avoid raising a ValueError in the center of mass calculation if the user only specifies half life which we want to disregard in this branch.